### ISSUE TITLE
Rule suggestion: `vue/valid-next-tick`

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#1399]
**Please describe what the rule should do:**
Calling `Vue.nextTick` or `vm.$nextTick` without passing a callback and without awaiting the returned Promise is likely a mistake (probably a missing `await`).

**What category should the rule belong to?**
[ ] Enforces code style (layout)
[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
import { nextTick as nt } from 'vue';

// BAD:
nt();
Vue.nextTick();
this.$nextTick();
nt;
Vue.nextTick;
this.$nextTick;

// GOOD:
await nt();
await Vue.nextTick();
await this.$nextTick();
nt().then(callback);
Vue.nextTick().then(callback);
this.$nextTick().then(callback);
nt(callback);
Vue.nextTick(callback);
this.$nextTick(callback);
```

**Additional context**

- [`Vue.nextTick` API in Vue 2](https://vuejs.org/v2/api/#Vue-nextTick)
- [`vm.$nextTick` API in Vue 2](https://vuejs.org/v2/api/#vm-nextTick)
- [Global API Treeshaking](https://v3.vuejs.org/guide/migration/global-api-treeshaking.html)
- [Global `nextTick` API in Vue 3](https://v3.vuejs.org/api/global-api.html#nexttick)
- [Instance `$nextTick` API in Vue 3](https://v3.vuejs.org/api/instance-methods.html#nexttick)

Should #1318 be part of this rule (with option `enforceStyle: 'promise'` or `enforceStyle: 'callback'`)?

**Comment by ota-meshi** [comment]
Thank you for suggesting this rule. I like this rule!
I think it makes sense that it's a different rule than `vue/next-tick-style` rule.
